src/client/auth/mod.rs

```diff
@@ -108,6 +108,13 @@ impl AuthMechanism {
         }
     }
 
+    pub(crate) fn is_scram(&self) -> bool {
```

Comment:
We need to add an extra field to the speculative auth document we send to the server when SCRAM is used, which makes having this helper useful